app/view/toolbar/ParallelLineToolbar.js

```diff
@@ -0,0 +1,165 @@
+         */
+        offsetUnit: 'meters'
+    },
+
+    items: [{
+        xtype: 'numberfield',
+        bind: {
+            fieldLabel: '{offsetLabel}',
+            value: '{offset}',
+        },
+        listeners: {
+            change: function () {
+                var toolbar = this.up('cmv_parallel_line_toolbar');
+
+                var vm = toolbar.getViewModel();
+                if (!vm) {
+                    return;
+                }
+                var parallelFeature = vm.get('parallelFeature');
+                if (!parallelFeature) {
+                    return;
+                }
+
+                toolbar.createParallelFeature();
+            }
+        }
+    }, {
+        glyph: 'ea50@font-gis',
+        bind: {
+            tooltip: '{parallelTooltip}',
+            disabled: '{!feature}'
+        },
+        handler: function() {
+            var toolbar = this.up('cmv_parallel_line_toolbar');
+            toolbar.createParallelFeature();
+        }
+    }],
+
+    applyFeature: function (feature) {
+        if (!feature) {
+            return;
+        }
+        var geometry = feature.getGeometry();
+        if (!geometry) {
+            console.warn('Feature has no geometry.');
+            return;
+        }
+
+        var geomType = geometry.getType();
+        if (geomType !== 'LineString' && geomType !== 'MultiLineString') {
+            console.warn('Unsupported feature geometry. Geometry must be of type "LineString" or "MultiLineString".');
+            return;
+        }
+        return feature;
+    },
+
+    updateFeature: function (newFeature) {
+        var vm = this.getViewModel();
+        if (!vm) {
+            return;
+        }
+        vm.set('feature', newFeature);
+        vm.set('parallelFeature', null);
+    },
+
+    applyOffsetUnit: function (offsetUnit) {
+        if (!offsetUnit) {
+            console.warn('offsetUnit is empty or undefined');
+            return;
+        }
+        var supportedUnits = [
+            'degrees', 'radians', 'miles', 'kilometers', 'inches',
+            'yards', 'meters'
+        ];
+        if (supportedUnits.indexOf(offsetUnit) === -1) {
+            console.warn(
+                'Provided offsetUnit not supported. Must be one of: ' + supportedUnits
+            );
+            return;
+        }
+        return offsetUnit;
+    },
+
+    /**
+     * Creates a parallel line for the given feature with given offset.
+     *
+     * Fires the parallelLineCreated event.
+     *
+     * @returns {void}
+     */
+    createParallelFeature: function() {
+        var vm = this.getViewModel();
+        if (!vm) {
+            return;
+        }
+        var feature = vm.get('feature');
+        if (!feature) {
+            return;
+        }
+
+        var offset = vm.get('offset');
+        if (offset === null) {
+            return;
+        }
+
+        var offsetUnit = this.getOffsetUnit();
+        var map = BasiGX.view.component.Map.guess().getMap();
```

Comment:
just wondering if we could move the part, which is not dependent on the component, to a util for "turf" functionality. It would be easier for testing and might be reused somewhere else.

Reply:
Sure, I can refactor this and create a util for that

Reply:
done in https://github.com/compassinformatics/cpsi-mapview/pull/577/commits/75cf1cc77e4a8316bb0557e9e67644d7ce15d8eb